### PR TITLE
Mariacarlina/fixing issues

### PR DESCRIPTION
**Pull request by mariacarlinahernandez**

### CODE REVIEW

File: src/UbidotsYUN.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    _server = server;
+    _dsName = "YUN";
+    _dsTag = "YUN";
+    val = (Value *)malloc(MAX_VALUES*sizeof(Value));
+}
+/** 
+ * This function set a data source tag
+ * @arg tag the tag to set
+ */
+void Ubidots::setDataSourceTag(char *tag) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: change this for setDeviceLabel()

File: src/UbidotsYUN.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    _dsTag = tag;
+}
+/** 
+ * This function set a data source name
+ * @arg name the name to set
+ */
+void Ubidots::setDataSourceName(char *name) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: change this for setDeviceName(). Put this method and setDataSourceTag() as deprecated. Do not forget retrocompatibility

File: src/UbidotsYUN.cpp [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+ */
+float Ubidots::parseValue(char *body) {
+    String raw;
+    char *pch;
+    float num;
+    char reply[25];
+    uint8_t bodyPosinit = 0;
+    uint8_t bodyPosend = 0;
+    pch = strstr(body, "\"value\":");
+    raw = String(pch);
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: do not use String objects

File: src/UbidotsYUN.cpp [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+ * @arg deviceLabel the device label where your variable is located
+ * @arg variableLabel the variable label where you will get the data
+ * @return num the data that you get from the Ubidots API
+ */
+float Ubidots::getValue(char* deviceLabel, char* variableLabel) {
+    const char* url = "http://things.ubidots.com/api/v1.6/devices";
+    Process _client;
+    float num;
+    char buffer[400];
+    char c[400];
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: Use friendly variables names, i.e server_response[]

File: src/UbidotsYUN.cpp [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    while (!_client.available() && timeout < 10000) {
+        timeout++;
+        delay(1);
+    }
+    while (_client.available() && i < 400) {
+        c[i] = _client.read();
+        i++;
+    }
+    c[i] = '\0';
+    while (_client.available()) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: this loop is useless

File: src/UbidotsYUN.cpp [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    }
+}
+/** 
+ * This function is to send all data saved in add function
+ */
+void Ubidots::sendAll() {
+    Process _client;
+    int i;
+    int timeout = 0;
+    String value;
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: do not use strings

File: src/UbidotsYUN.cpp [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    while (_client.running() && timeout < 10000) {
+        timeout++;
+        delay(1);
+    }
+    Serial.flush();
+    currentValue = 0;
+}
+// Old functions of the library
+
+bool Ubidots::saveValue(String id, float value) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: print a deprecated method warning

File: src/UbidotsYUN.h [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    char  *context;
+    float idValue;
+} Value;
+
+class Ubidots {
+ public:
+      Ubidots(char* token, char* server = SERVER);
+      void init();
+      float getValue(char* deviceLabel, char* variableLabel);
+      void add(char* id, float value, char* ctext = NULL);
```
Comment (jotathebest) [inline src/UbidotsYUN.h]: Modify this method to send values with custom timestamp

File: README.md [hunk 9]
```diff
@@ -69,49 +88,92 @@ Put your Ubidots token and variable ID where indicated, as well as the WiFi sett
...
+ * Main Functions
+ ****************************************/
 void setup() {
   client.init();
   Serial.begin(9600);
 }
 
 void loop() {
-  float value = client.getValue(ID);
+  float value = client.getValue(DEVICE_LABEL, VARIABLE_LABEL);
```
Comment (jotathebest) [inline README.md]: The method should check if the server answer is OK or not.

File: examples/UbidotsSendValueWithTimestamp/UbidotsSendValueWithTimestamp.ino [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+ ****************************************/
+#include <UbidotsYUN.h>
+
+/****************************************
+ * Define Constants
+ ****************************************/
+#define TOKEN "YOUR_TOKEN_HERE"
+#define VARIABLE_LABEL "temperature"  // Change for your variable label desired
+
+long unsigned timestamp = 1429272000; // init timestamp value
```
Comment (jotathebest) [inline examples/UbidotsSendValueWithTimestamp/UbidotsSendValueWithTimestamp.ino]: Add here that the timestamp should be in seconds

File: src/UbidotsYUN.cpp [hunk 11]
```diff
@@ -59,27 +62,8 @@ void Ubidots::init() {
     Bridge.begin();
     digitalWrite(13, HIGH);
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: why do you turn on this output? It should be available for the user

File: src/UbidotsYUN.cpp [hunk 12]
```diff
@@ -89,13 +73,12 @@ float Ubidots::getValue(char* deviceLabel, char* variableLabel) {
     const char* url = "http://things.ubidots.com/api/v1.6/devices";
     Process _client;
     float num;
-    char buffer[400];
+    char request[400];
     char c[400];
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: Again, use friendly names for the local variables

File: src/UbidotsYUN.cpp [hunk 13]
```diff
@@ -196,3 +184,19 @@ bool Ubidots::saveValue(String id, float value) {
     }
     Serial.flush();
 }
+
+/**
+ * This function set a data source tag
+ * @arg tag the tag to set
+ */
+void Ubidots::setDataSourceTag(char *tag) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: Add depŕecated warnings

File: src/UbidotsYUN.cpp [hunk 14]
```diff
@@ -196,3 +184,19 @@ bool Ubidots::saveValue(String id, float value) {
...
+    return setDeviceLabel(tag);
+}
+
+/**
+ * This function set a data source name
+ * @arg name the name to set
+ */
+void Ubidots::setDataSourceName(char *name) {
```
Comment (jotathebest) [inline src/UbidotsYUN.cpp]: Add deprecated warnings

File: library.properties [hunk 15]
```diff
@@ -0,0 +1,9 @@
+name=Ubidots Arduino YUN
+version=2.0.0
+author=Ubidots <devel@ubidots.com>, Mateo Velez
```
Comment (woakas) [inline library.properties]: Change to Devel Team <devel@ubidots.com>, Mateo Velez <devel@ubidots.com>

File: src/UbidotsYUN.h [hunk 16]
```diff
@@ -31,33 +32,34 @@ Made by Mateo Velez - Metavix for Ubidots Inc
...
-      void setDataSourceName(char *name);
+      float getValue(char* deviceLabel, char* variableLabel);
+      void add(char* id, float value, char* ctext = NULL, long unsigned timestamp_val = NULL);
+      void setDeviceLabel(char *deviceLabel);
+      void setDeviceName(char *deviceName);
       void sendAll();
       // Old functions
       bool saveValue(String id, float value);
-
+      void setDataSourceTag(char *tag); // Deprecated
```
Comment (woakas) [inline src/UbidotsYUN.h]: Remove this functions.

**Pull request merged**